ec2/spark_ec2.py

```diff
@@ -871,10 +898,11 @@ def deploy_files(conn, root_dir, opts, master_nodes, slave_nodes, modules):
 
     if "." in opts.spark_version:
         # Pre-built Spark deploy
-        spark_v = get_validate_spark_version(opts.spark_version, opts.spark_git_repo)
+        (spark_v, tachyon_v) = get_spark_tachyon_version(opts)
```

Comment:
Why are we removing the validation of the Spark version here? And why are we ignoring the user-specified GitHub repo?

This change doesn't look correct to me. Did I misunderstand something?